services/business/istio_validations_test.go

```diff
@@ -0,0 +1,229 @@
+package business
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/mock"
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/kiali/kiali/kubernetes"
+	"github.com/kiali/kiali/kubernetes/kubetest"
+)
+
+func TestServiceWellRouteRuleValidation(t *testing.T) {
+	assert := assert.New(t)
+
+	// Setup mocks
+	vs := mockValidationService(fakeIstioObjects())
+	validations, _ := vs.GetServiceValidations("bookinfo", "reviews")
+
+	// Well configured object
+	validObject := validations["reviews-well"]
+	assert.NotEmpty(validObject)
+	assert.Equal(validObject.Name, "reviews-well")
+	assert.Equal(validObject.ObjectType, "routerule")
+	assert.Equal(validObject.Valid, true)
+
+	// Assert checks
+	assert.Len(validObject.Checks, 0)
+}
+
+func TestServiceMultipleChecks(t *testing.T) {
+	assert := assert.New(t)
+
+	// Setup mocks
+	vs := mockValidationService(fakeMultipleChecks())
+	validations, _ := vs.GetServiceValidations("bookinfo", "reviews")
+
+	// wrong weight'ed route rule
+	invalidObject := validations["reviews-multiple"]
+	assert.NotEmpty(invalidObject)
+	assert.Equal(invalidObject.Name, "reviews-multiple")
+	assert.Equal(invalidObject.ObjectType, "routerule")
+	assert.Equal(invalidObject.Valid, false)
+
+	checks := invalidObject.Checks
+	assert.Len(checks, 2)
+
+	assert.Equal(checks[0].Message, "Weight should be between 0 and 100")
+	assert.Equal(checks[0].Severity, "error")
+	assert.Equal(checks[0].Path, "spec/route/weight/155")
```

Comment:
Looking this one makes me wonder how to deal with collections.
For example, we might have several weights, so one option could be to maintain this, and let the user to iterate with all the weight to check the offending one, or just to mark a [0], [1], etc.

I like the default one, but just thinking loud.

Reply:
That is a good point. When trying to define this PATH field, I was thinking of that moment when you are printing a specific field into the UI. At that moment, you know exactly where you are: 'spec/route/weight' and you know the value: 155. Then it is easy to check if that specific field has a error/warning.
I am unsure how this would work for showing errors/warnings into an ACE Editor. Does it worth to check now how to approach this before having it into master?

Reply:
btw, I like your proposal as well.